src/main/groovy/edu/oregonstate/mist/trailsapi/resources/TrailResources.groovy

```diff
@@ -81,4 +81,25 @@ public class TrailsResource extends Resource {
 	Boolean trailValidator(Trail trail) {
 		trail.name && trail.zipCode && trail.difficulty
 	}
+
+	@GET
+	@Produces(MediaType.APPLICATION_JSON)
+	Response getTrail (@QueryParam("name") Optional<String> name,
+					@QueryParam("difficulty") Optional<String> difficulty,
+					@QueryParam("mostDifficult") Optional<String> mostDifficult,
+					@QueryParam("leastDifficult") Optional<String> leastDifficult,
+					@QueryParam("zipCode") Optional<Integer> zipCode,
+					@QueryParam("smallDrop") Optional<Boolean> smallDrop,
+					@QueryParam("largeDrop") Optional<Boolean> largeDrop,
+					@QueryParam("woodRide") Optional<Boolean> woodRide,
+					@QueryParam("skinny") Optional<Boolean> skinny,
+					@QueryParam("largeJump") Optional<Boolean> largeJump,
+					@QueryParam("smallJump") Optional<Boolean> smallJump,
+					@QueryParam("gap") Optional<Boolean> gap) {
+			List<Trail> trails = trailDAO.getTrailByQuery(name.or(""), difficulty.or(""),
+				mostDifficult.or(""), leastDifficult.or(""), zipCode.orNull(), smallDrop.orNull(),
+				largeDrop.orNull(), woodRide.orNull(), skinny.orNull(), largeJump.orNull(), smallJump.orNull(),
+				gap.orNull())
+			ok(trails).build()
```

Comment:
The JSON this method should return should be a ResultObject, where the data object is an array of resource objects: https://github.com/osu-mist/mtb-trails-api/blob/79df752e58ae9f8b99788f2424d3710552faf3ca/swagger.yaml#L295-L309